core/modules/search/search.install

```diff
@@ -144,6 +144,37 @@ function search_schema() {
   return $schema;
 }
 
+/**
+ * Save content types to index.
+ */
+function search_install() {
+  // Node types are created in the install profile after search module is enabled,
+  // so we preset the initial search configurations here.
+   $config = config('search.settings');
+  $config->set('search_minimum_word_size', 3);
+  $config->set('search_overlap_cjk', 1);
+  $config->set('search_cron_limit', 100);
+  $config->set('search_tag_weights', array(
+    'h1' => 25,
+    'h2' => 18,
+    'h3' => 15,
+    'h4' => 12,
+    'h5' => 9,
+    'h6' => 6,
+    'u' => 3,
+    'b' => 3,
+    'i' => 3,
+    'strong' => 3,
+    'em' => 3,
+    'a' => 10
+  ));
+  $config->set('search_active_modules', array('node', 'user'));
+  $config->set('search_and_or_limit', 7);
+  $config->set('search_default_module', 'node');
+  $config->set('content_types', array('page' => 'page', 'post' => 'post'));
+  $config->save();
```

Comment:
We shouldn't need to set these in hook_install(), they should just be in the default config provided by search module. And if `hook_node_type_insert()` is implemented, we don't need to specify particular node types here.